core/common/src/Segment.kt

```diff
@@ -345,14 +345,20 @@ public class Segment {
         limit += srcEndOffset - srcStartOffset
     }
 
-    @PublishedApi
-    @get:JvmSynthetic
-    internal val size: Int
+    /**
+     * The number of readable bytes contained in this segment.
+     *
+     * @sample kotlinx.io.samples.unsafe.UnsafeBufferOperationsSamples.readUleb128
+     */
+    public val size: Int
```

Comment:
Perhaps, it should be `Long`.

Reply:
But then both `pos` and `limit` have to be updated too.

Reply:
And the reason why we might need `Long` properties here is supporting MemorySegmented backed segments on JVM (and that's nowhere near).

Reply:
`MemorySegment`s can be sliced which should be a cheap operation. If one came in over 2^31 in length, it could be sliced into multiple segments.